### PR TITLE
Bump HLS to 1.7.0.0

### PR DESCRIPTION
**Pull request by paolino**
- [x] I have cherry picked anviking commit about hls bump from 1.5.0.0 to 1.7.0.0
- [x] Suspended  CI checks for HLS until version 1.7.0.1 is out

### Issue Number

ADP-1916

### CODE REVIEW

File: nix/overlays/build-tools.nix [hunk 1]
```diff
@@ -33,12 +33,13 @@ pkgs: super: let
...
     cabal-install.version           = "3.4.0.0";
     haskell-language-server = {
-      version = "1.5.0.0";
+      version = "1.7.0.0";
       modules = [{ reinstallableLibGhc = false; }];
     };
     hie-bios = {
       modules = [{ reinstallableLibGhc = false; }];
     };
+    hie-bios = {};
```
Comment (Anviking) [inline nix/overlays/build-tools.nix]: Be careful not to override the intentionally set `[{ reinstallableLibGhc = false; }];`
```suggestion
```

File: cabal.project.old [hunk 2]
```diff
@@ -0,0 +1,398 @@
+--------------------------------------------------------------------------------
+-- Cabal project file
```
Comment (Anviking) [inline cabal.project.old]: Shouldn't be committed I presume

File: check-cabal-project.sh [hunk 3]
```diff
@@ -0,0 +1 @@
+/nix/store/19m3ybns2whd3lnc4jydzd1zhw93xq0r-check-cabal-project
```
Comment (Anviking) [inline check-cabal-project.sh]: Shouldn't be committed. Perhaps we should add it to .gitignore though.

**Review by Anviking — APPROVED** [review]

File: cabal.project.old (hunk 2 above)
Comment (paolino) [reply cabal.project.old]: done

File: check-cabal-project.sh (hunk 3 above)
Comment (paolino) [reply check-cabal-project.sh]: done

**Review by Anviking — APPROVED** [review]
👍

**Pull request merged**